cmd/thanos/rule.go

```diff
@@ -120,8 +120,11 @@ func registerRule(app *extkingpin.App) {
 	httpMethod := cmd.Flag("query.http-method", "HTTP method to use when sending queries. Possible options: [GET, POST]").
 		Default("POST").Enum("GET", "POST")
 
+	// The native Go resolver fails to parse compressed responses for SRV records.
+	// The miekgdns resolver doesn't suffer the same issue hence defaulting to it instead.
+	// See https://bugzilla.redhat.com/show_bug.cgi?id=1953518
 	dnsSDResolver := cmd.Flag("query.sd-dns-resolver", "Resolver to use. Possible options: [golang, miekgdns]").
```

Comment:
Why not pass it as flag in https://github.com/openshift/cluster-monitoring-operator/blob/master/jsonnet/thanos-ruler.libsonnet#L376-L393 ? We already have this pattern for querier in https://github.com/openshift/cluster-monitoring-operator/blob/master/jsonnet/thanos-querier.libsonnet#L345

Reply:
we don't have a way to pass the flag with the Prometheus operator (at least I've tried and haven't found).

Reply:
ack.

/lgtm